### PR TITLE
Rewrite activity feed query to use the new Keepscussion repos

### PR DESCRIPTION
**Pull request by leogrim**
@ryanpbrewster @andrewconner

### CODE REVIEW

**Comment by andrewconner** [comment]
:+1:

File: kifi-backend/modules/shoebox/app/com/keepit/model/KeepRepo.scala [hunk 1]
```diff
@@ -80,6 +78,8 @@ class KeepRepoImpl @Inject() (
     val clock: Clock,
     libraryMembershipRepo: LibraryMembershipRepo, // implicit dependency on this repo via a plain SQL query getRecentKeeps
     organizationMembershipRepo: OrganizationMembershipRepo, // implicit dependency on this repo via a plain SQL query getRecentKeeps
+    keepToLibraryRepo: KeepToLibraryRepo, // implicit dependency on this repo via a plain SQL query getRecentKeeps
+    keepToUserRepo: KeepToUserRepo, // implicit dependency on this repo via a plain SQL query getRecentKeeps
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/model/KeepRepo.scala:82]: I like this, nice comments.

File: kifi-backend/modules/shoebox/app/com/keepit/model/KeepRepo.scala [hunk 2]
```diff
@@ -544,46 +544,58 @@ class KeepRepoImpl @Inject() (
...
+    val keepsFromUser = s"""SELECT ktu.keep_id as id, ktu.added_at as first_added_at FROM $ktu_WHERE_THIS_USER AND $added_at_BEFORE"""
+
+    val keepsAndFirstAddedAt = sql"""
+      SELECT k.id as keep_id, min(k.first_added_at) as added_at
+      FROM ((#$keepsFromLibraries) UNION (#$keepsFromOrganizations) UNION (#$keepsFromUser)) as k
+      GROUP BY k.id
+      HAVING #$added_at_AFTER AND #$added_at_BEFORE
+      ORDER BY added_at DESC, keep_id DESC
+      LIMIT $limit
+    """.as[(Id[Keep], DateTime)].list
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/model/KeepRepo.scala:594]: Do we need this to be `(Id[Keep], DateTime)`? Can't we just `SELECT k.id as keep_id` and call it a day?

Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/model/KeepRepo.scala:594]: Oh, I see, we're ordering by an aggregate so you want to save it. I don't think that's strictly necessary, though.

Can you use `SELECT k.id as keep_id ... ORDER BY min(k.first_added_at)`?

Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/model/KeepRepo.scala:594]: All super low priority changes.

**Pull request merged**